test/testdrive/kafka-avro-debezium-sinks-reingest.td

```diff
@@ -0,0 +1,153 @@
+# by the Apache License, Version 2.0.
+
+# Test behavior that is specific to Kafka Avro Sinks with ENVELOPE DEBEZIUM
+
+# Test a basic sink with multiple rows.
+
+> CREATE VIEW data (a, b) AS VALUES (1, 1), (2, 1), (3, 1), (1, 2)
+
+> CREATE SINK data_sink FROM data
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'data-sink'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+$ kafka-verify format=avro sink=materialize.public.data_sink sort-messages=true
+{"before": null, "after": {"row": {"a": 1, "b": 1}}}
+{"before": null, "after": {"row": {"a": 1, "b": 2}}}
+{"before": null, "after": {"row": {"a": 2, "b": 1}}}
+{"before": null, "after": {"row": {"a": 3, "b": 1}}}
+
+# More complex sinks, with multiple keys and/or a consistency topic. We test
+# all the possible combinations of user-specified sink key and
+# natural (primary) relation key.
+
+$ set schema={
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+      {
+        "name": "before",
+        "type": [
+          {
+            "name": "row",
+            "type": "record",
+            "fields": [
+              {"name": "a", "type": "long"},
+              {"name": "b", "type": "long"}
+            ]
+          },
+          "null"
+        ]
+      },
+      { "name": "after", "type": ["row", "null"] }
+    ]
+  }
+
+$ set trxschemakey={
+      "name": "io.debezium.connector.common.TransactionMetadataKey",
+      "type": "record",
+      "fields": [
+          {
+              "name": "id",
+              "type": "string"
+          }
+      ]
+  }
+
+$ set trxschema={
+    "type":"record", "name":"TransactionMetadataValue", "namespace":"io.debezium.connector.common",
+    "fields":[
+    {"name":"status","type":"string"},
+    {"name":"id","type": "string"},
+    {"name": "event_count",
+    "type": ["null", "long"],
+    "default": null
+    },
+    {"name":"data_collections","type":["null",{"type":"array",
+    "items": {"type":"record",
+    "name":"ConnectDefault",
+    "namespace":"io.confluent.connect.Avro",
+    "fields": [ {
+    "name": "data_collection",
+    "type": "string"
+    },
+    {
+    "name": "event_count",
+    "type": "long" }]}}],
+    "default": null}],
+    "connect.name": "io.debezium.connector.common.TransactionMetadataValue"
+    }
+
+$ kafka-create-topic topic=consistency
+$ kafka-create-topic topic=input
+
+# first create all the sinks, then ingest data, to ensure that
+# input is processed in consistency batches and not all at once
+
+> CREATE MATERIALIZED SOURCE input
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-input-${testdrive.seed}'
+    WITH (consistency_topic = 'testdrive-consistency-${testdrive.seed}')
+  FORMAT AVRO USING SCHEMA '${schema}' ENVELOPE DEBEZIUM
+
+> CREATE SINK exactly_once_sink FROM input
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'exactly-once-sink-${testdrive.seed}'
+# exactly_once = true -> causes test failure, false -> test works
```

Comment:
Switching `exactly_once` around causes this test to fail. I'm wondering if it is because I'm missing something...